mne/morph.py

```diff
@@ -0,0 +1,953 @@
+    ``fsaverage_sym`` subject is included with FreeSurfer > 5.1 and can be
+    obtained as described `here
+    <http://surfer.nmr.mgh.harvard.edu/fswiki/Xhemi>`_. For statistical
+    comparisons between hemispheres, use of the symmetric ``fsaverage_sym``
+    model is recommended to minimize bias [1]_.
+
+    References
+    ----------
+    .. [1] Greve D. N., Van der Haegen L., Cai Q., Stufflebeam S., Sabuncu M.
+           R., Fischl B., Brysbaert M.
+           A Surface-based Analysis of Language Lateralization and Cortical
+           Asymmetry. Journal of Cognitive Neuroscience 25(9), 1477-1492, 2013.
+    """
+    logger.info('Computing morph matrix...')
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+
+    tris = _get_subject_sphere_tris(subject_from, subjects_dir)
+    maps = read_morph_map(subject_from, subject_to, subjects_dir, xhemi)
+
+    if xhemi:
+        hemi_indexes = [(0, 1), (1, 0)]
+    else:
+        hemi_indexes = [(0, 0), (1, 1)]
+
+    morph = []
+    for hemi_from, hemi_to in hemi_indexes:
+        idx_use = vertices_from[hemi_from]
+        if len(idx_use) == 0:
+            continue
+        e = mesh_edges(tris[hemi_from])
+        e.data[e.data == 2] = 1
+        n_vertices = e.shape[0]
+        e = e + sparse.eye(n_vertices, n_vertices)
+        m = sparse.eye(len(idx_use), len(idx_use), format='csr')
+        mm = _morph_buffer(m, idx_use, e, smooth, n_vertices,
+                           vertices_to[hemi_to], maps[hemi_from], warn=warn)
+        morph.append(mm)
+
+    if len(morph) == 0:
+        raise ValueError("Empty morph-matrix")
+    elif len(morph) == 1:
+        morph = morph[0]
+    else:
+        morph = sparse_block_diag(morph, format='csr')
+    logger.info('[done]')
+    return morph
+
+
+###############################################################################
+# Apply pre-computed morph
+
+@verbose
+def _apply_morph(stc_from, morph, verbose=None):
+    """Morph a source estimate from one subject to another.
+
+    Parameters
+    ----------
+    stc_from : VolSourceEstimate | VectorSourceEstimate | SourceEstimate
+        Data to be morphed
+    morph : instance of SourceMorph
+        SourceMorph obtained from SourceMorph(src, subject_from, subject_to)
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    stc_to : VolSourceEstimate | VectorSourceEstimate | SourceEstimate
+        Source estimate for the destination subject.
+    """
+    if morph.kind == 'volume':
+
+        if not has_dipy():
+            raise ImportError(
+                'DiPy (Python) must be correctly installed and accessible '
+                'from Python')
+
+        from dipy.align.imwarp import DiffeomorphicMap
+        from dipy.align.imaffine import AffineMap
+        from dipy.align.reslice import reslice
```

Comment:
let's assume that `dipy` and `nibabel` is required for morphing and check once

This seems like way too much duplication

Reply:
well it's not required for surface morphs... - but I'll make it a general check, which will cause an ImportError any time the user tries to create a ``SourceMorph`` in case either nibabel or dipy is not present... Is that what you meant?